### PR TITLE
*: add some metrics for rocksdb

### PR DESCRIPTION
**Pull request by DorianZheng**
## What have you changed? (mandatory)

add metric for ingest sst
add metric for number of files at each level at each cf

## What are the type of the changes? (mandatory)

Improvement

## How has this PR been tested? (mandatory)

Just add metric

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

Yes, I will add it later

### CODE REVIEW

File: src/util/rocksdb/engine_metrics.rs [hunk 1]
```diff
@@ -955,6 +956,20 @@ pub fn flush_engine_properties(engine: &DB, name: &str) {
                     .set(v);
             }
         }
+
+        // Num files at levels
+        let opts = engine.get_options_cf(handle);
+        for level in 0..opts.get_num_levels() {
+            let level_str = level.to_string();
```
Comment (huachaohuang) [inline src/util/rocksdb/engine_metrics.rs]: You can use `format!` here, have a look at `rocksdb::get_engine_compression_ratio_at_level`.

**Comment by huachaohuang** [comment]
CI failed, please make format and run tests before creating a PR.

File: src/raftstore/store/metrics.rs [hunk 2]
```diff
@@ -214,4 +214,11 @@ lazy_static! {
             "tikv_raftstore_leader_missing",
             "Total number of leader missed region"
         ).unwrap();
+
+    pub static ref INGEST_SST_DURATION_SECONDS: Histogram =
+        register_histogram!(
+            "tikv_ingest_sst_duration_seconds",
```
Comment (overvenus) [inline src/raftstore/store/metrics.rs]: Sounds like missing a subsystem `tikv_(subsystem)_ingest_sst_duration_seconds`.

File: src/util/rocksdb/engine_metrics.rs [hunk 3]
```diff
@@ -1215,6 +1228,11 @@ lazy_static! {
         "Oldest unreleased snapshot duration in seconds",
         &["db"]
     ).unwrap();
+    pub static ref STORE_ENGINE_NUM_FILES_AT_LEVEL_VEC: IntGaugeVec = register_int_gauge_vec!(
```
Comment (siddontang) [inline src/util/rocksdb/engine_metrics.rs]: leave a blank line

File: src/util/rocksdb/engine_metrics.rs [hunk 4]
```diff
@@ -955,6 +956,18 @@ pub fn flush_engine_properties(engine: &DB, name: &str) {
                     .set(v);
             }
         }
+
+        // Num files at levels
+        let opts = engine.get_options_cf(handle);
+        for level in 0..opts.get_num_levels() {
+            let prop = format!("{}{}", ROCKSDB_COMPRESSION_RATIO_AT_LEVEL, level);
```
[the remark this answers is not in the document]
Comment (DorianZheng) [reply src/util/rocksdb/engine_metrics.rs]: 原来我一commit到我的branch就会出现在这里，233333，我改一下

File: src/util/rocksdb/engine_metrics.rs (hunk 3 above)
Comment (DorianZheng) [reply src/util/rocksdb/engine_metrics.rs]: ok

**Comment by siddontang** [comment]
@DorianZheng  Please sign the CLA

LGTM

File: src/raftstore/store/metrics.rs [hunk 5]
```diff
@@ -214,4 +214,11 @@ lazy_static! {
             "tikv_raftstore_leader_missing",
             "Total number of leader missed region"
         ).unwrap();
+
+    pub static ref INGEST_SST_DURATION_SECONDS: Histogram =
+        register_histogram!(
+            "tikv_snapshot_ingest_sst_duration_seconds",
+            "Bucketed histogram of duration of rocksdb ingestion",
```
Comment (Hoverbear) [inline src/raftstore/store/metrics.rs]: "Bucketed histogram of rocksdb ingestion durations"

File: src/raftstore/store/snap.rs [hunk 6]
```diff
@@ -972,6 +972,7 @@ impl Snapshot for Snap {
                 let mut file = box_try!(File::open(&cf_file.path));
                 apply_plain_cf_file(&mut BufReader::new(file), &options, cf_handle)?;
             } else {
+                let _timer = INGEST_SST_DURATION_SECONDS.start_coarse_timer();
```
Comment (Hoverbear) [inline src/raftstore/store/snap.rs]: I see you start this timer but I don't see you do other things to the timer (such as stop it).

Comment (siddontang) [reply src/raftstore/store/snap.rs]: we will calculate the elapsed time in Drop

**Comment by siddontang** [comment]
ping @DorianZheng

File: src/raftstore/store/metrics.rs (hunk 5 above)
Comment (DorianZheng) [reply src/raftstore/store/metrics.rs]: ok

**Pull request merged**